datajoint/blob.py

```diff
@@ -201,41 +212,95 @@ def __str__(self):
         return str(self._blob[self.pos:])
 
 
-def pack(obj):
-    """
-    Packs an object into a blob to be compatible with mym.mex
+def pack(obj, compress=True):
+    blob = b"mYm\0"
+    blob += pack_obj(obj)
 
-    :param obj: object to be packed
-    :type obj: numpy.ndarray
-    """
-    if not isinstance(obj, np.ndarray):
-        raise DataJointError("Only numpy arrays can be saved in blobs")
+    if compress:
+        compressed = b'ZL123\0' + np.uint64(len(blob)).tostring() + zlib.compress(blob)
+        if len(compressed) < len(blob):
+            blob = compressed
+
+    return blob
+
+
+def pack_obj(obj):
+    blob = b''
+    if isinstance(obj, np.ndarray):
+        blob += pack_array(obj)
+    elif isinstance(obj, Mapping):  # TODO: check if this is a good inheritance check for dict etc.
+        blob += pack_dict(obj)
+    elif isinstance(obj, str):
+        blob += pack_array(np.array(obj, dtype=np.dtype('c')))
+    elif isinstance(obj, Iterable):
```

Comment:
If `obj` is an iterator, then `np.array` will not exhaust it. Replace with `list(obj)`.